tests_async/unit/requests/test_download.py

```diff
@@ -88,87 +83,125 @@ async def test__write_to_stream_with_hash_check_fail(self):
-        
+        transport.request = mock.AsyncMock(spec=["__call__"], return_value=mockResponse)
 
         assert not download.finished
-        ret_val = await download.consume(transport)
+
+        if timeout is not None:
+            ret_val = await download.consume(transport, timeout=timeout)
+        else:
+            ret_val = await download.consume(transport)
+
         assert ret_val is transport.request.return_value
 
-        called_kwargs = {u"data": None, u"headers": download._headers}
+        called_kwargs = {
+            u"data": None,
+            u"headers": download._headers,
+            u"timeout": EXPECTED_TIMEOUT if timeout is None else timeout}
+
         if chunks:
             assert stream is not None
             called_kwargs[u"stream"] = True
         transport.request.assert_called_once_with(
-            u"GET", EXAMPLE_URL, timeout=EXPECTED_TIMEOUT, **called_kwargs
+            u"GET", sync_test.EXAMPLE_URL, **called_kwargs
         )
 
         range_bytes = u"bytes={:d}-{:d}".format(0, end)
         assert download._headers[u"range"] == range_bytes
         assert download.finished
 
         return transport
-    
+
     @pytest.mark.asyncio
     async def test_consume(self):
-        await self._consume_helper()
+        await self._consume_helper(timeout=14.7)
 
+    @pytest.mark.parametrize("checksum", ["md5", "crc32c", None])
     @pytest.mark.asyncio
-    async def test_consume_with_stream(self):
+    async def test_consume_with_stream(self, checksum):
         stream = io.BytesIO()
         chunks = (b"up down ", b"charlie ", b"brown")
-        transport = await self._consume_helper(stream=stream, chunks=chunks)
+        await self._consume_helper(stream=stream, chunks=chunks, checksum=checksum)
 
         assert stream.getvalue() == b"".join(chunks)
 
         # Check mocks.
-        response = transport.request.return_value
+        # response = transport.request.return_value
 
+    @pytest.mark.parametrize("checksum", ["md5", "crc32c"])
     @pytest.mark.asyncio
-    async def test_consume_with_stream_hash_check_success(self):
+    async def test_consume_with_stream_hash_check_success(self, checksum):
         stream = io.BytesIO()
         chunks = (b"up down ", b"charlie ", b"brown")
-        header_value = u"md5=JvS1wjMvfbCXgEGeaJJLDQ=="
-        headers = {download_mod._HASH_HEADER: header_value}
-        transport = await self._consume_helper(
-            stream=stream, chunks=chunks, response_headers=headers
+        header_value = u"crc32c=UNIQxg==,md5=JvS1wjMvfbCXgEGeaJJLDQ=="
+        headers = {_helpers._HASH_HEADER: header_value}
+        await self._consume_helper(
+            stream=stream, chunks=chunks, response_headers=headers, checksum=checksum
         )
 
         assert stream.getvalue() == b"".join(chunks)
 
         # Check mocks.
-        response = transport.request.return_value
+        # response = transport.request.return_value
```

Comment:
should this be commented out?

Reply:
Added back, changed test and passing.

Reply:
Diverging from the sync implementation here - the sync implementation asserts that we called on response.raw.stream which we don't have in aiohttp responses. Should I add a TODO on this?